packages/arb-token-bridge-ui/src/components/common/Dialog.tsx

```diff
@@ -84,105 +83,116 @@ export function useDialog(params?: UseDialogParams): UseDialogResult {
-  const isCustom = props.isCustom || false
-  const closeable = props.closeable || true
+  const isFooterHidden = props.isFooterHidden || false
+  const closeable = props.closeable ?? true
   const className = props.className || ''
   const cancelButtonRef = useRef(null)
 
+  // separate state to track transition state and have a smooth exit animation
+  const [isClosing, setIsClosing] = useState(false)
+  const [isConfirmed, setIsConfirmed] = useState(false)
+
+  const handleCloseStart = useCallback(
+    (confirmed: boolean) => {
+      if (!confirmed && !closeable) {
+        return
+      }
+
+      setIsConfirmed(confirmed)
+      setIsClosing(true)
+    },
+    [setIsClosing, setIsConfirmed, closeable]
+  )
+
+  const handleCloseEnd = useCallback(() => {
+    setIsClosing(false)
+    props.onClose(isConfirmed)
+  }, [props, isConfirmed, setIsClosing])
+
   return (
-    <Transition
-      appear
-      show={props.isOpen}
-      as={Fragment}
-      {...getTransitionProps('normal')}
-    >
+    <Transition show={props.isOpen && !isClosing} as={Fragment}>
       <HeadlessUIDialog
-        static
         as="div"
         open={props.isOpen}
         initialFocus={props.initialFocus || cancelButtonRef}
-        onClose={() => props.onClose(false)}
+        onClose={() => handleCloseStart(false)}
         className="fixed inset-0 z-50 flex text-white md:items-center md:justify-center"
       >
         <Transition.Child
           as={Fragment}
-          enter="ease-out duration-300"
+          enter="ease-out duration-[400ms]"
           enterFrom="opacity-0"
           enterTo="opacity-80"
-          leave="ease-in duration-300"
+          leave="ease-in duration-200"
           leaveFrom="opacity-80"
           leaveTo="opacity-0"
         >
           <div className="fixed inset-0 bg-black" aria-hidden="true" />
         </Transition.Child>
         <Transition.Child
           as={Fragment}
-          enter="ease-out duration-300"
+          enter="ease-out duration-[400ms]"
           enterFrom="opacity-0 scale-95"
           enterTo="opacity-100 scale-100"
-          leave="ease-in duration-300"
+          leave="ease-in duration-200"
           leaveFrom="opacity-100 scale-100"
           leaveTo="opacity-0 scale-95"
+          afterLeave={handleCloseEnd}
         >
-          <HeadlessUIDialog.Panel
-            className={twMerge(
-              'z-10 max-h-screen w-full overflow-y-auto border border-gray-dark bg-gray-1 md:max-h-[calc(100vh-80px)] md:w-auto md:rounded',
-              className
-            )}
-          >
-            {isCustom ? (
-              props.children
-            ) : (
-              <>
-                <div className="flex items-center justify-between px-5 py-5 pt-3">
-                  <HeadlessUIDialog.Title className="text-xl">
-                    {props.title}
-                  </HeadlessUIDialog.Title>
+          <div className={twMerge('w-screen md:max-w-[727px]', className)}>
```

Comment:
move the `z-10` from the line below to here

but is there a reason this div is needed? these tailwind classes can be applied to the `HeadlessUIDialog.Panel` directly?

Reply:
removed the div above